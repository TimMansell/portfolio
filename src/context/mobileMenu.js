import React, { useState, createContext } from 'react';
import PropTypes from 'prop-types';

export const MenuContext = createContext();

export const MenuContextProvider = ({ children }) => {
  const [isMobileMenu, setIsMobileMenu] = useState(false);

  return (
    <MenuContext.Provider value={[isMobileMenu, setIsMobileMenu]}>
      {children}
    </MenuContext.Provider>
  );
};

MenuContextProvider.propTypes = {
  children: PropTypes.node.isRequired
};
