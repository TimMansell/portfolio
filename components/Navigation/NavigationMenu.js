import React, {useContext} from "react";
import {MenuContext} from "../../context/mobileMenu";
import classnames from "classnames";
import {Link} from "react-scroll";

import useInMobileViewport from "hooks/useInMobileViewport";
import styles from "./NavigationMenu.module.scss";
import navigationJson from "./json/navigation.json";

import {combineWords, upperCaseWords} from "./helpers";

import {useIsFlexGapSupported} from "hooks/useFlexGap";

export const NavigationMenu = () => {
  const hasFlexGap = useIsFlexGapSupported();
  const [isMobileMenu, setIsMobileMenu] = useContext(MenuContext);
  const inViewport = useInMobileViewport();

  const menuClasses = classnames(styles.menu, {
    [styles.menuIsActive]: isMobileMenu && inViewport,
    [styles.flexGap]: hasFlexGap,
  });

  return (
    <ul className={menuClasses} data-e2e="navigation">
      {navigationJson.map((navigation, index) => (
        <li className={styles.menuItem} key={index}>
          <Link
            data-e2e={`navigation-${combineWords(navigation)}`}
            className={styles.link}
            activeClass={styles.linkIsActive}
            to={combineWords(navigation)}
            smooth={true}
            spy={true}
            duration={500}
            onClick={() => setIsMobileMenu(false)}
          >
            {upperCaseWords(navigation)}
          </Link>
        </li>
      ))}
    </ul>
  );
};

export default NavigationMenu;
