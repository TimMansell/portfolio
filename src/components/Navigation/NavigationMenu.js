import React, { useContext } from 'react';
import { MenuContext } from '../../context/mobileMenu';
import classnames from 'classnames';
import { Link } from 'react-scroll';

import useInMobileViewport from 'hooks/useInMobileViewport';
import styles from './NavigationMenu.module.scss';
import navigationJson from './json/navigation.json';

import { combineWords, upperCaseWords } from './helpers';

export const NavigationMenu = () => {
  const [isMobileMenu, setIsMobileMenu] = useContext(MenuContext);
  const inViewport = useInMobileViewport();

  const menuClasses = classnames(styles.menu, {
    [styles.menuIsActive]: isMobileMenu && inViewport,
  });

  return (
    <ul className={menuClasses}>
      {navigationJson.map((navigation, index) => (
        <li className={styles.menuItem} key={index}>
          <Link
            data-e2e={`navigation-${combineWords(navigation)}`}
            className={styles.link}
            activeClass={styles.linkIsActive}
            to={combineWords(navigation)}
            smooth={true}
            spy={true}
            duration={500}
            onClick={() => setIsMobileMenu(false)}
          >
            {upperCaseWords(navigation)}
          </Link>
        </li>
      ))}
    </ul>
  );
};

export default NavigationMenu;
