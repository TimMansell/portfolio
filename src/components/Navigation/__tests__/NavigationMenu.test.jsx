import { it, describe, expect } from 'vitest';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import NavigationMenu from '../NavigationMenu';
import { MenuContext } from '../../../context/mobileMenu';

vi.mock('../json/navigation.json', () => ({
  default: ['item 1', 'item 2', 'item 3'],
}));

const MockComponent = () => (
  <MenuContext.Provider value={[false, vi.fn()]}>
    <NavigationMenu />
  </MenuContext.Provider>
);

describe('NavigationMenu', () => {
  it('should render my component', () => {
    // eslint-disable-next-line
    const wrapper = shallow(<MockComponent />);
  });

  it('should match snapshot', () => {
    const snapshot = renderer.create(<MockComponent />).toJSON();

    expect(snapshot).toMatchSnapshot();
  });
});
