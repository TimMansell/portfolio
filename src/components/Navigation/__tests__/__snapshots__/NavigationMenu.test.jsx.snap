// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavigationMenu > should match snapshot 1`] = `
<ul
  className="_menu_2b860d"
  data-e2e="navigation"
>
  <li
    className="_menuItem_2b860d"
  >
    <a
      className="_link_2b860d"
      data-e2e="navigation-item-1"
      onClick={[Function]}
    >
      Item 1
    </a>
  </li>
  <li
    className="_menuItem_2b860d"
  >
    <a
      className="_link_2b860d"
      data-e2e="navigation-item-2"
      onClick={[Function]}
    >
      Item 2
    </a>
  </li>
  <li
    className="_menuItem_2b860d"
  >
    <a
      className="_link_2b860d"
      data-e2e="navigation-item-3"
      onClick={[Function]}
    >
      Item 3
    </a>
  </li>
</ul>
`;
